backend/lib/endpoints/posts.js

```diff
@@ -131,6 +132,22 @@ async function routes(app) {
       };
       /* eslint-enable sort-keys */
 
+      // if location is defined, use simple regex text query, in order to use $geoNear
+      if (location && keywords) {
+        const keywordsRegex = new RegExp(
+          keywords.split(/[ ,]+/).join("|"),
```

Comment:
This will search for each keyword as with a logical `or` right?
I'm not sure if we need to do this, this may make it difficult to search for bigrams and trigrams (compound words) where their unigrams are common. Maybe just searching for the keyword without splitting will get better results.

Anyway, I think we do need some normalization on the search term (`keywords` variable here) like lowercasing, maybe encoding to utf-8 to remove weird characters... 
I think that can be helpful for the regex and the full-text search